pkg/controller/bundle_sync_task.go

```diff
@@ -42,7 +44,13 @@ type bundleSyncTask struct {
 // READY state might mean something different for each resource type. For a Custom Resource it may mean
 // that a field "State" in the Status of the resource is set to "Ready". It is customizable via
 // annotations with some defaults.
-func (st *bundleSyncTask) process() (retriableError bool, e error) {
+func (st *bundleSyncTask) processNormal() (retriableError bool, e error) {
+	// If the "deleteResources" finalizer is missing, add it and finish the processing iteration
+	if !hasDeleteResourcesFinalizer(st.bundle) {
+		st.newFinalizers = addDeleteResourcesFinalizer(st.bundle.GetFinalizers())
```

Comment:
Is this racy? Reacting to the delete change at the same time the gc can?

Shouldn't we just add the finalizer to bundles during first create?